### ISSUE TITLE
Error installing the new alternate area for Soubar.

### ISSUE THREAD

**Issue by PixelsnPolygons** [SpellholdStudios/Shadows_over_Soubar#44]
I am trying to include the alternate Soubar area for the Shadows over Soubar mod in my EET playthrough, but I keep running into this error whilst installing that component. Anyone ever see that before? Is there a prerequisite I am missing that enables the DELETE_EXISTING_ANIMATIONS macro?

Thank you for creating and maintaining so many wonderful mods!

//ERROR: [SoS/Are/Alt/AR4230.are] -> [override/AR4230.are] Patching Failed (COPY) (Failure("Unknown macro: DELETE_EXISTING_ANIMATIONS"))
//Stopping installation because of error.
//ERROR Installing [Alternative Soubar area (by Cuttooth)], rolling back to previous state
//Will uninstall  50 files for [SoS\SoS.tp2] component 4.
//Uninstalled     50 files for [SoS\SoS.tp2] component 4.
//ERROR: Failure("Unknown macro: DELETE_EXISTING_ANIMATIONS")

[WeiDU-BGEE.log](https://github.com/user-attachments/files/15935397/WeiDU-BGEE.log)
[WeiDU.log](https://github.com/user-attachments/files/15935399/WeiDU.log)

**Comment by Roberciiik** [comment]
I don't know what happened here yet. But it seems to be related to @weigo87's latest changes. Please try removing from sos.tp2 file this part of the code 
`PATCH_INCLUDE ~%MOD_FOLDER%/lib/install-nightmaps.tpa~`
in the alternate Soubar component. Let me know if this helps with the issue.

**Comment by Roberciiik** [comment]
@PixelsnPolygons 
I updated the code. Could you please download and check the latest master branch? If that helps, I will submit a new version. Thanks!

BTW, `DELETE_EXISTING_ANIMATIONS` macro is included in the SoS code base, so it is really strange that the file is not loaded during your installation. I did test installation of EET + SoS and the mod installed fine.

**Comment by weigo87** [comment]
Found the problem.
The DELETE_EXISTING_ANIMATIONS macro can be found in the macro.tpa and this was installed with the main component. @PixelsnPolygons has probably installed the alternative Soubar component separately and thus the sos.tp2 did not load the macro.tpa and the macro was no longer available. 
@Roberciiik If you install both components in one go, the macro is still in the memory and can be used.
I have now packed the line "INCLUDE ~%MOD_FOLDER%/lib/macro.tpa~" into the ALWAYS part, where it is automatically reloaded for each component and it works again.

**Comment by Roberciiik** [comment]
@weigo87 
Thanks for finding a better solution for loading the macro. Unfortunately, the game crashes when "nightmaps" are installed for alternative Soubar.ARE, as it seems that `are_patch/ar4230.tpa` file is shared between old Soubar (Beregost based) and a new one, but their animations are not intact, so there are some errors in ARE file causing BG2:EE to CTD. I am removing this patch for now until proper solution found.

**Comment by PixelsnPolygons** [comment]
@weigo87 I grabbed the new master package and was able to successfully install the alternate Soubar area component. A bit of additional context for you: my initial installation attempt was part of a larger install sequence using Project Infinity in which the SoS core component and the alternate Soubar area component were consecutive. Not sure, but it's possible PI closes out each WeiDU instance after a successful installation, even if the next component being installed is part of the same mod? That might explain why the macro definition was dropped from memory when the INCLUDE reference to the macro TPA/TPH was not in ALWAYS.

@Roberciiik I also launched a new SoA campaign after the installation and did not get a CTD, to further test your "nightmaps" workaround.

Thanks for the quick responses - you guys rule!

PS: If this means what I think it means, then I am going to be very happy!
![image](https://github.com/SpellholdStudios/Shadows_over_Soubar/assets/172719315/61488503-5000-47d8-bd80-63cb0556c98b)
